thread/src/main/kotlin/io/github/paulgriffith/kindling/thread/MultiThreadView.kt

```diff
@@ -280,24 +269,22 @@ class MultiThreadView(
             }
         }
 
+        // populate initial state of all the filter lists
+        visibleThreadDumps = threadDumps
+
         poolList.checkBoxListSelectionModel.bind()
         stateList.checkBoxListSelectionModel.bind()
         systemList.checkBoxListSelectionModel.bind()
+
         threadDumpCheckboxList.checkBoxListSelectionModel.addListSelectionListener { event ->
             if (!event.valueIsAdjusting) {
                 listModelsAdjusting = true
 
-                val selectedThreadDumps = threadDumps.filterIndexed { index, _ ->
-                    index + 1 in (event.source as CheckBoxListSelectionModel).selectedIndices
-                }
-                allThreads = selectedThreadDumps.flatMap { it.threads }
-
-                poolList.setModel(allThreads.groupingBy(Thread::pool).eachCount())
-                stateList.setModel(allThreads.groupingBy(Thread::state).eachCount())
-                systemList.setModel(allThreads.groupingBy(Thread::system).eachCount())
+                val selectedIndices = (event.source as CheckBoxListSelectionModel).selectedIndices
 
+                val selectedThreadDumps = threadDumps.slice(selectedIndices.map { it - 1 })
+                visibleThreadDumps = selectedThreadDumps
```

Comment:
Now this is the only place that filters the list of visible thread dumps. This updates visibleThreadDumps, which will aggregate again, kick off the reaction in `set()` there, and ultimately end up calling `updateData()` again that way. Again - bonus of not repeating filtering/updating logic in multiple places.